addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4701,14 +4689,22 @@ registry.ImageOptimize = ImageHandlerOption.extend({
             } else {
                 // If the preview mode === false we want to save the colors
                 // as the user chose their shape
-                await this._applyShapeAndColors(previewMode === false);
+                await this._applyShapeAndColors(saveData);
+                if (saveData) {
+                    img.dataset.originalMimetype = img.dataset.mimetype;
+                    img.dataset.mimetype = 'image/svg+xml';
+                }
             }
         } else {
             // Re-applying the modifications and deleting the shapes
             img.src = await applyModifications(img);
             delete img.dataset.shape;
             delete img.dataset.shapeColors;
             delete img.dataset.fileName;
+            if (saveData) {
+                img.dataset.mimetype = img.dataset.originalMimetype;
+                delete img.dataset.originalMimetype;
+            }
```

Comment:
1- Let's take the example of `willStart`, a saved PNG image with a shape will have `dataset = {..., mimetype: 'image/svg+xml', originalMimetype: 'image/png'}` (which make sense)

But on `willStart`, we should set the original value (`'image/png'`) on `img.dataset.mimetype` to have image options available **(see ImageHandlerOption > _renderCustomXML)**. Then, we should keep the right mimetype value since the image will be saved by the `saveModifiedImages` method...

The `dataset.mimetype` is also used by `image_processing` > `applyModifications` to apply options on images, so without the code in `willStart`, we won't be able to update correctly images inside shapes.

I updated the code to get the right mimetype for image when updated by options using a `_getImageMimetype` method.
It's better this way and we don't need to add the original mimetype on `willStart` :slightly_smiling_face: